components/ChartLegend.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div class="legend-container m-1">
+    <div class="legend-element d-flex flex-column gap-1">
+      <div
+        v-for="item in items"
+        :key="item.label"
+        :class="`legend-item legend-item-${item.shape}`"
+        :style="legendItemStyle(item)"
+      >
+        <i :style="iStyle(item)" />
+        <span
+          class="level"
+          :style="{ lineHeight: `${props.rowHeightRem}rem` }"
+        >
+          {{ item.label }}
+        </span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { type LegendItem, LegendShape } from "./utils/charts";
+
+const props = defineProps<{
+  items: LegendItem[]
+  rowHeightRem: number
```

Comment:
Yes, this is because time series is showing two rows, and costs is showing 3. Arguably this could be calculated in a way that involves a division by number of rows (and incorporating some padding calculations to account for fencepost errors) but I didn't want to get too bogged down with that.